python/eos/parameter.py

```diff
@@ -99,24 +99,40 @@ def _repr_html_(self):
 
 
     @staticmethod
-    def FromWCxf(wc):
+    def FromWCxf(w):
+        """
+        Imports Wilson coefficients into an eos.Parameters object from a wilson.Wilson object.
+
+        :param w: WET Wilson coefficients in the EOS basis.
+        :type w: wilson.Wilson
+        """
+
+        # extract wc data
+        wc = w.wc
+
+        if wc.basis != 'EOS':
+            raise ValueError('Wilson coefficients must be passed to this function in the EOS basis')
+
+        # Needs to be revisited, once EOS support WET-4 or WET-3 bases
+        if wc.scale != 4.2:
+            raise ValueError('Wilson coefficients must be passed to this function at the reference scale 4.2 GeV')
 
         # the following coefficients are treated as real-valued in EOS
         real_coeffs = [
             'c1', 'c2', 'c3', 'c4', 'c5', 'c6', 'c8', 'c8\''
         ]
 
         parameters = _Parameters.Defaults()
-        for name in wc.dict:
-            qn     = eos.QualifiedName(name)
+        for name, value in wc.dict.items():
+            qn     = QualifiedName(name)
             prefix = qn.prefix_part()
             coeff  = qn.name_part()
-            value  = wc.dict[name]
-            if (not value.imag == 0) and (coeff in real_coeffs):
-                raise ValueError('WC {0} does not support non-zero imaginary part'.format(name))
```

Comment:
I remember that Peter Stangl mentioned small imaginary part due to running and matching https://github.com/eos/eos/issues/335#issuecomment-874747524

Reply:
Indeed, see below.